webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteActions.js

```diff
@@ -0,0 +1,76 @@
+import URI from 'urijs';
+import API from '../../API';
+import { STATUS } from '../../constants';
+import {
+  AUTO_COMPLETE_REQUEST,
+  AUTO_COMPLETE_SUCCESS,
+  AUTO_COMPLETE_FAILURE,
+  AUTO_COMPLETE_RESET,
+  TRIGGERS,
+} from './AutoCompleteConstants';
+
+export const getResults =
+(path, searchQuery, controller, trigger) => (dispatch) => {
+  const loadNextResults = trigger === TRIGGERS.ITEM_SELECT ? '+' : '';
+  const url = `${path}${URI.encodeQuery(searchQuery)}${loadNextResults}`;
+
+  // Starting a request
+  dispatch({
+    type: AUTO_COMPLETE_REQUEST,
+    payload: {
+      controller,
+      searchQuery,
+      status: STATUS.PENDING,
+      trigger,
+    },
+  });
+
+  // Making an API call
+  return API.get(url)
+    .then(({ data }) => {
+      const { error } = data[0] || {};
+      // The API can return errors too.
+      if (error) {
+        throw error;
```

Comment:
I don't know what is better but I like doing `Promise.reject(error)`.
@amirfefer @sharvit any preferences? Otherwise, lets keep `throw error`.